File: apps/jetbrains/plugin/src/test/kotlin/io/autocomplete/bridge/SettingsExportTest.kt
package io.autocomplete.bridge

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Test

class SettingsExportTest {
    @Test
    fun redactOmitsSecretMarkersAndExtraHeadersButKeepsPortableProfileFields() {
        val exported =
            SettingsExport.redact(
                mapOf(
                    "schemaVersion" to 1,
                    "profiles" to
                        listOf(
                            mapOf(
                                "id" to "profile-1",
                                "name" to "Gateway",
                                "baseUrl" to "https://example.test/v1",
                                "apiKey" to "must-not-escape",
                                "hasApiKey" to true,
                                "authHeaderTemplate" to "X-API-Key: must-not-escape",
                                "extraHeadersJson" to "{\"X-API-Key\":\"must-not-escape\"}",
                            ),
                        ),
                ),
            )

        @Suppress("UNCHECKED_CAST")
        val profile = (exported["profiles"] as List<Map<String, Any?>>).single()
        assertEquals("profile-1", profile["id"])
        assertEquals("Gateway", profile["name"])
        assertEquals("https://example.test/v1", profile["baseUrl"])
        assertFalse(profile.containsKey("apiKey"))
        assertFalse(profile.containsKey("hasApiKey"))
        assertFalse(profile.containsKey("authHeaderTemplate"))
        assertFalse(profile.containsKey("extraHeadersJson"))
    }
}

File: apps/vscode/extension/src/config/settingsExport.test.ts
import assert from "node:assert/strict";
import { describe, it } from "node:test";
import { redactSettingsExport } from "./settingsExport.js";

describe("redactSettingsExport", () => {
  it("removes credential-capable fields without mutating the local snapshot", () => {
    const snapshot = {
      schemaVersion: 1,
      enabled: true,
      profiles: [
        {
          id: "profile-1",
          name: "Gateway",
          baseUrl: "https://example.test/v1",
          model: "model-a",
          hasApiKey: true,
          apiKey: "must-not-escape",
          authHeaderTemplate: "X-API-Key: must-not-escape",
          extraHeadersJson: '{"X-API-Key":"must-not-escape"}',
          promptTemplate: "CHAT",
        },
      ],
    };

    const exported = redactSettingsExport(snapshot);

    assert.deepEqual(exported, {
      schemaVersion: 1,
      enabled: true,
      profiles: [
        {
          id: "profile-1",
          name: "Gateway",
          baseUrl: "https://example.test/v1",
          model: "model-a",
          promptTemplate: "CHAT",
        },
      ],
    });
    assert.equal(snapshot.profiles[0].authHeaderTemplate, "X-API-Key: must-not-escape");
    assert.equal(snapshot.profiles[0].extraHeadersJson, '{"X-API-Key":"must-not-escape"}');
  });
});
